railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -95,11 +95,9 @@ def bin
     end
 
     def bin_when_updating
-      bin_yarn_exist = File.exist?("bin/yarn")
-
       bin
 
-      if options[:api] && !bin_yarn_exist
+      if options[:skip_yarn]
```

Comment:
When `options[:api]` is truthy, `option[:skip_yarn]` is also truthy. 
https://github.com/rails/rails/blob/42b9e7e50c084e119a679cf155b70b5efc4d36ff/railties/lib/rails/generators/rails/app/app_generator.rb#L281-L283